File: frontend/src/types/order.ts
export type OrderStatus =
  | 'PENDING'
  | 'CONFIRMED'
  | 'SHIPPING'
  | 'DELIVERED'
  | 'CANCELLED';

export interface OrderItem {
  productId: string;
  productName: string;
  productImage: string;
  brand: string;
  price: number;
  quantity: number;
}

export interface Order {
  id: string;
  userId: string;
  email: string;
  customerName: string;
  phone: string;
  address: string;
  city: string;
  district: string;
  ward: string;
  note?: string;
  paymentMethod: string;
  status: OrderStatus;
  items: OrderItem[];
  subtotal: number;
  shippingFee: number;
  total: number;
  createdAt: string;
  paymentStatus?: string;
  momoTransId?: string;
}

export interface CreateOrderPayload {
  email: string;
  customerName: string;
  phone: string;
  address: string;
  city: string;
  district: string;
  ward: string;
  note?: string;
  paymentMethod: string;
  items: OrderItem[];
}

export const ORDER_STATUS_LABEL: Record<OrderStatus, string> = {
  PENDING: 'Chờ xác nhận',
  CONFIRMED: 'Đã xác nhận',
  SHIPPING: 'Đang giao',
  DELIVERED: 'Đã giao',
  CANCELLED: 'Đã hủy',
};

export const ORDER_STATUS_COLOR: Record<OrderStatus, string> = {
  PENDING: 'bg-yellow-50 text-yellow-700',
  CONFIRMED: 'bg-blue-50 text-blue-700',
  SHIPPING: 'bg-purple-50 text-purple-700',
  DELIVERED: 'bg-green-50 text-green-700',
  CANCELLED: 'bg-red-50 text-red-700',
};

File: nhom12/src/main/java/nhom12/example/nhom12/model/Order.java
package nhom12.example.nhom12.model;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import nhom12.example.nhom12.model.enums.OrderStatus;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Order extends BaseDocument {

  @Indexed private String userId;

  private String email;
  private String customerName;
  private String phone;
  private String address;
  private String city;
  private String district;
  private String ward;
  private String note;
  private String paymentMethod;
  private OrderStatus status;
  private List<OrderItem> items;
  private double subtotal;
  private double shippingFee;
  private double total;

  // Payment tracking
  private String paymentStatus; // PENDING, PAID, FAILED
  private String momoTransId;
}
